src/components/GroupModal/GroupModalContent.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { FC, useRef, ReactNode } from "react";
+import { useIntersection } from "react-use";
+import CheckBox from "../checkbox/Checkbox";
+import { useText } from "../../core/utils/text";
+import { getRenewableMaterials } from "../../core/utils/helpers/general";
+
+interface GroupModalContentProps {
+  amountOfSelectableMaterials: number;
+  selectableMaterials: string[];
+  selectedMaterials: string[];
+  buttonComponent: ReactNode;
+  selectMaterials: (materialIds: string[]) => void;
+  children: ReactNode;
+}
+
+const GroupModalContent: FC<GroupModalContentProps> = ({
+  amountOfSelectableMaterials,
+  selectableMaterials,
+  selectedMaterials,
+  selectMaterials,
+  buttonComponent,
+  children
+}) => {
+  const t = useText();
+
+  const intersectionRef = useRef(null);
+  const { isIntersecting: isVisible } = useIntersection(intersectionRef, {
+    threshold: 0
+  }) || { isIntersecting: false };
```

Comment:
I don't understand why this is necessary.
The GroupModalContent component will only be visible the modal is rendering its children, right? Otherwise it is returning null. So as far as I see it the behaviour is build in.

Reply:
I did this a while ago, so honestly I am not entirely sure. but. I think useIntersection returns null or isIntersecting (and other stuff), so the isIntersection : false was a sane backup. but I've changed it now.